concurrency/MpmcQueue.h

```diff
@@ -249,13 +282,21 @@ class MpmcQueue
 
     // Layout below must be maintained for performance. The non atomic member variables may not be place before read
     // cursor.
-    std::atomic<VersionedIndex> _readCursor;
-    uint32_t _cacheLineSeparator1[14];
-    std::atomic<VersionedIndex> _writeCursor;
-    uint32_t _cacheLineSeparator2[14];
-    const uint32_t _entriesPerCacheline;
-    const uint32_t _capacity;
-    const size_t _blockSize;
+    alignas(64) std::atomic<VersionedIndex> _readCursor;
+
+    // Special Cell state for when queue has zero size
+    // Null Entry is not written and it is only read it when queue is zero size. So it can live in same cache line of
+    // _readCursor without performance impact
+    const EntrySate _nullEntry = CellState::State::nullSlot;
+
+    alignas(64) std::atomic<VersionedIndex> _writeCursor;
+
+    // Although they are const. Cache can be invalidated for the reader when _writeCursor
```

Comment:
// place _capacity in next cacheline to avoid interference with writeCursor on size() and capacity()